pkg/services/publicdashboards/database/database_test.go

```diff
@@ -270,9 +270,70 @@ func TestIntegrationFindByDashboardUid(t *testing.T) {
 	t.Run("returns error when public dashboard doesn't exist", func(t *testing.T) {
 		setup()
 		pubdash, err := publicdashboardStore.FindByDashboardUid(context.Background(), 9, "fake-dashboard-uid")
-		require.Error(t, err)
-		require.Nil(t, pubdash)
-		assert.Equal(t, ErrPublicDashboardNotFound, err)
+		require.NoError(t, err)
+		assert.Nil(t, pubdash)
+	})
+}
+
+func TestIntegrationFindByAccessToken(t *testing.T) {
+	var sqlStore db.DB
+	var cfg *setting.Cfg
+	var dashboardStore *dashboardsDB.DashboardStore
+	var publicdashboardStore *PublicDashboardStoreImpl
+	var savedDashboard *models.Dashboard
+
+	setup := func() {
+		sqlStore, cfg = db.InitTestDBwithCfg(t)
+		dashboardStore = dashboardsDB.ProvideDashboardStore(sqlStore, cfg, featuremgmt.WithFeatures(), tagimpl.ProvideService(sqlStore, cfg))
+		publicdashboardStore = ProvideStore(sqlStore)
+		savedDashboard = insertTestDashboard(t, dashboardStore, "testDashie", 1, 0, true)
+	}
+
+	t.Run("returns isPublic and set dashboardUid and orgId", func(t *testing.T) {
+		setup()
+		savedPubdash := insertPublicDashboard(t, publicdashboardStore, savedDashboard.Uid, savedDashboard.OrgId, false)
+		pubdash, err := publicdashboardStore.FindByAccessToken(context.Background(), savedPubdash.AccessToken)
+		require.NoError(t, err)
+		assert.Equal(t, savedPubdash, pubdash)
+	})
+
+	t.Run("returns dashboard errDashboardIdentifierNotSet", func(t *testing.T) {
+		setup()
+		_, err := publicdashboardStore.FindByAccessToken(context.Background(), "")
+		require.Error(t, dashboards.ErrDashboardIdentifierNotSet, err)
```

Comment:
nolonger returning this error

Reply:
Yep, good catch. It didn't fail because it should be `require.Equal` instead of `require.Error`. Changing to assert `Nil` for both returning parameters.